static/app/components/nav/sidebar.tsx

```diff
@@ -89,37 +91,83 @@ const SidebarItemList = styled('ul')`
   }
 `;
 
-function SidebarItem({item}: {item: NavSidebarItem}) {
-  const location = useLocation();
-  const isActive = isNavItemActive(item, location);
-  const isSubmenuActive = isSubmenuItemActive(item, location);
-  const _to = resolveNavItemTo(item);
-  const linkProps = _to ? makeLinkPropsFromTo(_to) : {to: '#'};
+interface SidebarItemProps {
+  item: NavSidebarItem;
+}
+
+function SidebarItem({item}: SidebarItemProps) {
+  const to = resolveNavItemTo(item);
+  const SidebarChild = to ? SidebarLink : SidebarMenu;
 
   const FeatureGuard = item.feature ? Feature : Fragment;
   const featureGuardProps: any = item.feature ?? {};
 
   return (
     <FeatureGuard {...featureGuardProps}>
       <SidebarItemWrapper>
-        <SidebarLink
-          {...linkProps}
-          aria-current={isActive ? 'page' : undefined}
-          aria-selected={isActive || isSubmenuActive}
-        >
-          <InteractionStateLayer hasSelectedBackground={isActive || isSubmenuActive} />
+        <SidebarChild item={item} key={item.label}>
           {item.icon}
           <span>{item.label}</span>
-        </SidebarLink>
+        </SidebarChild>
       </SidebarItemWrapper>
     </FeatureGuard>
   );
 }
 
-const SidebarLink = styled(Link)`
+const NavLink = styled(Link)`
   position: relative;
 `;
 
+const NavButton = styled('button')`
+  border: none;
+  position: relative;
+  background: transparent;
+  min-width: 58px;
+
+  ${linkStyles}
+`;
+
+function SidebarLink({children, item}: SidebarItemProps & {children: React.ReactNode}) {
+  const location = useLocation();
+  const isActive = isNavItemActive(item, location);
+  const isSubmenuActive = isSubmenuItemActive(item, location);
+  const to = resolveNavItemTo(item);
+  const linkProps = makeLinkPropsFromTo(to!);
+
+  return (
+    <NavLink
+      {...linkProps}
+      className={isActive || isSubmenuActive ? 'active' : undefined}
+      aria-current={isActive ? 'page' : undefined}
+    >
+      <InteractionStateLayer hasSelectedBackground={isActive || isSubmenuActive} />
+      {children}
+    </NavLink>
+  );
+}
+
+function SidebarMenu({item, children}: SidebarItemProps & {children: React.ReactNode}) {
+  if (!item.dropdown) {
+    throw new Error(
+      `Nav item "${item.label}" must have either a \`dropdown\` or \`to\` value!`
+    );
+  }
```

Comment:
Should there be a similar check in SiedbarLink?

Reply:
Good call, added in [b5e7e44](https://github.com/getsentry/sentry/pull/79720/commits/b5e7e44da2e819b015e0e99f20db5aa8f52e1473)